src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt

```diff
@@ -238,6 +242,15 @@ fun processPathResolveVariants(lookup: ImplLookup, path: RsPath, isCompletion: B
     return processNestedScopesUpwards(path, processor, ns)
 }
 
+private inline fun <reified T : PsiElement> PsiElement.contextOfType(strict: Boolean = true): T? {
```

Comment:
Looks like we already have this method [here](https://github.com/intellij-rust/intellij-rust/blob/1a1ea675f6c72db2358845518c85986b879aa394/src/main/kotlin/org/rust/lang/core/psi/ext/PsiElement.kt#L35-L36)